src/openttd.cpp

```diff
@@ -498,6 +515,12 @@ struct AfterNewGRFScan : NewGRFScanCallback {
 			NetworkClientConnectGame(NetworkAddress(network_conn, rport), join_as, join_server_password, join_company_password);
 		}
 
+#if defined(UNIX)
+		if (_settings_client.gui.newgrf_developer_tools) {
```

Comment:
I think this test should be in the handler

Reply:
True. As written you need to quit and restart the game after enabling `newgrf_developer_tools` for the signal feature to be available.